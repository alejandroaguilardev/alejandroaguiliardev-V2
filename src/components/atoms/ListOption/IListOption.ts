export interface IListOption {
	title: string;
	list: { title: string; secondary?: string; type?: string }[];
}
