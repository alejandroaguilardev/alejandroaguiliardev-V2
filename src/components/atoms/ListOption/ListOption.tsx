import { Typography, List, ListItem, ListItemText, Divider, Box } from "@mui/material"
import { IListOption } from "./IListOption"

export const ListOption = ({ title, list }: IListOption) => {
    return (
        <>
            <Typography color='primary.contrastText' fontSize={16} fontWeight="bold">
                {title}
            </Typography>

            <List component="nav" aria-label="mailbox folders">
                {
                    list.map((data) => (
                        <Box key={data.title}>
                            <ListItem button>
                                <ListItemText primary={data.title} secondary={<Typography fontSize={13}>{data.secondary}</Typography>} sx={{ color: 'primary.contrastText' }} />
                            </ListItem>
                            <Divider light />
                        </Box>

                    ))
                }
            </List>
        </>
    )
}
