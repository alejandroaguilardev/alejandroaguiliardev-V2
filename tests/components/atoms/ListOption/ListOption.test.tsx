import { render, screen } from "@testing-library/react";
import { ListOption } from "../../../../src/components/atoms/ListOption/ListOption";

describe("ListOption", () => {
	const list = [
		{ title: "title one", secondary: "second Text" },
		{ title: "title SecondZ", secondary: "", type: "link" },
	];
	test("toMatchSnapshot ", () => {
		const { container } = render(<ListOption title="Title test" list={list} />);
		expect(container).toMatchSnapshot();
	});

	test("Props ", () => {
		render(<ListOption title="Title test" list={list} />);

		expect(screen.findByText(list[0].title)).toBeTruthy();
		expect(screen.findByText(list[0].secondary)).toBeTruthy();
		expect(screen.findByText(list[1].title)).toBeTruthy();
	});
});
