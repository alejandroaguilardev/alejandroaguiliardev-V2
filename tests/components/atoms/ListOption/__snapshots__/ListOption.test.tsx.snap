// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListOption toMatchSnapshot  1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1469zy3-MuiTypography-root"
  >
    Title test
  </p>
  <nav
    aria-label="mailbox folders"
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-1itc3p-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            title one
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-qybatc-MuiTypography-root"
          >
            second Text
          </p>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light css-xetjfm-MuiDivider-root"
      />
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-1itc3p-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            title SecondZ
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-qybatc-MuiTypography-root"
          />
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light css-xetjfm-MuiDivider-root"
      />
    </div>
  </nav>
</div>
`;

exports[`ListOption toMatchSnapshot 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1469zy3-MuiTypography-root"
  >
    Title test
  </p>
  <nav
    aria-label="mailbox folders"
    class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-1itc3p-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            title one
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-qybatc-MuiTypography-root"
          >
            second Text
          </p>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light css-xetjfm-MuiDivider-root"
      />
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-bshv44-MuiButtonBase-root-MuiListItem-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-1itc3p-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            title SecondZ
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-qybatc-MuiTypography-root"
          />
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light css-xetjfm-MuiDivider-root"
      />
    </div>
  </nav>
</div>
`;
